Lib/test/support/pty_helper.py

```diff
@@ -58,3 +58,23 @@ def terminate(proc):
                         input = b""  # Stop writing
                     if not input:
                         sel.modify(master, selectors.EVENT_READ)
+
+
+######################################################################
+## Fake stdin (for testing interactive debugging)
+######################################################################
+
+class FakeInput:
+    """
+    A fake input stream for pdb'sx interactive debugger.  Whenever a
+    line is read, print it (to simulate the user typing it), and then
+    return it.  The set of lines to return is specified in the
+    constructor; they should not have trailing newlines.
+    """
+    def __init__(self, lines):
+        self.lines = lines
+
+    def readline(self):
+        line = self.lines.pop(0)
```

Comment:
Calling `list.pop(0)` frequently is inefficient. Maybe the constructor should create a `collections.deque()` and call `self.lines.popleft()` instead?

Reply:
It is usually used on cases like `sys.stdin = FakeInput(['next', 'print(x)', 'continue'])`, I don't think it matters much. But, I can do this if you want :)